andela/activo-api
# #159557184 An endpoint to update a role

#### What does this PR do?
Allows a user to edit a role

#### Description of Task to be completed?
As a developer, I should be able to edit a role  by making a patch request to `/roles/<role_id>`

#### How should this be manually tested?
1. Do a `git fetch origin ft-update-role-159557184` to fetch the remote branch
2. Do a `git checkout ft-update-role-159557184` to checkout to your branch
3. Do a `pipenv shell` to activate your virtual environment
4.  Run your app using  `flask run`
5. Using postman, specify your token and make a:
- `PATCH` request to `/roles/<role_id>` with the correct payload as specified below:
```
{
  "title":  "role name" ,
  "description": "role description"
}
```
role_id should be the role id of an existing role. A `200` response is returned on success together with the details for the newly updated role.
- `PATCH` request to the `/roles/<role_id>` route using the same payload and a `409` error response is returned
- `PATCH` request to the `/roles/<role_id>` route a role_id containing symbols such as @# and a `400` error response is returned.
- `PATCH` request to the `/roles/<role_id>` without a bearer token and a `401` error response is returned

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#159557184](https://www.pivotaltracker.com/story/show/159557184)

#### What pivotal tracker stories does this block?
None

#### What pivotal tracker stories are blocking this?
None

### Postman Documentation
[Activo update role](https://www.getpostman.com/collections/36fd4cdd11521d81fa38)

#### Screenshots (if appropriate)
Successful update of a role
<img width="1126" alt="screen shot 2018-08-16 at 16 43 43" src="https://user-images.githubusercontent.com/17172747/44213468-7242e680-a176-11e8-8b5e-477ecf486a3b.png">

When duplicate role exists
<img width="1116" alt="screen shot 2018-08-13 at 16 55 04" src="https://user-images.githubusercontent.com/17172747/44036164-d48726da-9f19-11e8-94c2-956c2adb9d97.png">

When no token supplied with header
<img width="1116" alt="screen shot 2018-08-13 at 16 57 27" src="https://user-images.githubusercontent.com/17172747/44036275-0c5c5616-9f1a-11e8-928d-b3425e9a1f55.png">

Invalid id parameter. For example, id containing #@* 
<img width="1110" alt="screen shot 2018-08-13 at 16 58 30" src="https://user-images.githubusercontent.com/17172747/44036318-2ccc2912-9f1a-11e8-8e0f-bdb3d897ed07.png">
